src/arrayMethodJoin.js

```diff
@@ -4,9 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let arr = '';
+    const separatorCopy = separator + '';
+    // console.log(this);
+
+    for (let i = 0; i < this.length; i++) {
+      arr += (this[i] === undefined || this[i] === null ? '' : this[i])
```

Comment:
if you want your code to be cleaner use if/else. For it is very difficult to understand what kind of logic you wanted to put into the ternary operator.